tests/test_cluster_ref_key.cc

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file   test_cluster_ref_key.cc
+ *
+ * @author Alexander Goscinski <alexander.goscinski@epfl.ch>
+ * @date   10 Thu 2019
+ *
+ * @brief  tests for the cluster ref key layer computation utilities
+ *
+ * Copyright  2019 Alexander Goscinski, COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include "test_cluster_ref_key.hh"
+
+namespace rascal {
+  BOOST_AUTO_TEST_SUITE(test_cluster_ref_key);
+
+  /**
+   * tests the get_min_layer function
+   */
+  BOOST_AUTO_TEST_CASE(test_get_min_layer) {
+    // LayersByOrder using only one root manager
+    const size_t active_layer_root_manager{
+        get_min_layer<1>(typename std::index_sequence<0>{})};
+    BOOST_CHECK_EQUAL(0, active_layer_root_manager);
+    // LayersByOrder for `AdaptorFullNeighbourList`
+    const size_t active_layer_order_one_adaptor{
+        get_min_layer<1>(typename std::index_sequence<1, 0>{})};
+    BOOST_CHECK_EQUAL(1, active_layer_order_one_adaptor);
+    // LayersByOrder for stacking an `AdaptorMaxOrder` on an `AdaptorStrict`
+    const size_t active_layer_increase_max_order{
+        get_min_layer<3>(typename std::index_sequence<1, 1, 0>{})};
+    BOOST_CHECK_EQUAL(0, active_layer_increase_max_order);
+  }
+
+  /**
+   * tests the get_layer function
+   */
+  BOOST_FIXTURE_TEST_CASE(test_get_layer, LayerFixture) {
+    BOOST_CHECK(get_layer<1>(order_three_seq{}) == 1);
+    BOOST_CHECK(get_layer<2>(order_three_seq{}) == 1);
+    BOOST_CHECK(get_layer<3>(order_three_seq{}) == 0);
+  }
+
+  /**
+   * tests the get_layers function
+   */
+  BOOST_FIXTURE_TEST_CASE(test_get_layers, LayerFixture) {
+    std::array<size_t, 1> arr_order_one = get_layers(order_one_seq{});
+    std::array<size_t, 1> arr_order_one_ref = {1};
+    BOOST_CHECK(arr_order_one == arr_order_one_ref);
+
+    std::array<size_t, 2> arr_order_two = get_layers(order_two_seq{});
+    std::array<size_t, 2> arr_order_two_ref = {1, 1};
+    BOOST_CHECK(arr_order_two == arr_order_two_ref);
+
+    std::array<size_t, 3> arr_order_three = get_layers(order_three_seq{});
+    std::array<size_t, 3> arr_order_three_ref = {1, 1, 0};
+    BOOST_CHECK(arr_order_three == arr_order_three_ref);
+    print_index_sequence(std::make_index_sequence<5>{});
```

Comment:
```suggestion
     if (verbose) {
       print_index_sequence(std::make_index_sequence<5>{});
     }
```

Reply:
Plus a verbose flag at the top of the test.